docs_dev/releasing.md

```diff
@@ -160,14 +160,92 @@ Alternatively you can use the UI.
    ```
    * Set the tag to be the correct version for the tag.
 
-## Releasing a new version of the website
+## Versioning the website
 
-For each stable release, we also publish a corresponding version of the 
-Kubeflow [website](www.kubeflow.org). Each version of the website is generated 
-from a separate [branch](https://github.com/kubeflow/website/branches)
-of the kubeflow/website repository. 
+The main Kubeflow website at [www.kubeflow.org](www.kubeflow.org) points to the
+**master** branch of the `kubeflow/website` repo. Similarly, 
+[master.kubeflow.org](https://master.kubeflow.org/) also points to the master
+branch.
 
-We usually create the website branch for a version a few weeks after the 
+For each minor (X.Y) release, we also publish a corresponding version of the 
+website. Each version of the website is generated from a separate 
+[branch](https://github.com/kubeflow/website/branches)
+of the `kubeflow/website` repository. The naming convention is as follows:
+The site at `vX-Y.kubeflow.org` points to the release at vX.Y-branch.
+For example, the [documentation for v0.6](https://v0-6.kubeflow.org) is 
+maintained in the
+[v0.6-branch](https://github.com/kubeflow/website/tree/v0.6-branch).
+
+There are two release phases when we need to update the website:
+* Updating version numbers etc for an upcoming release (major, minor, or patch).
+* Creating a website branch for the latest major or minor release.
+
+### Updating version numbers etc for the upcoming release (major, minor, or patch)
+
+We usually start updating the website for the upcoming major/minor release when
+we've released a good RC for that release. At this point, it's useful for
+users to start experimenting with the upcoming release.
+
+When we're ready to start updating the docs for the upcoming version, we
+need to update various version numbers on the website, to ensure users are
+getting the correct installation instructions etc. Here are the steps to follow:
+
+1. Make sure the website branch for the current major/minor version has been 
```

Comment:
Its somewhat ambiguous which version current is referring to; i.e. is it the version already released or the version we are releasing.

My suggestion would be in the preamble to say something like suppose we are in the process of releasing 0.Y so the docs are currently serving version 0.X. 

Then you can refer to 0.Y and 0.X below and its clear what you are refering to.

Reply:
Or you can just say "if we are releasing 0.7 then make sure the branch for v0.6 exists" that would be consistent with what you have below.

Reply:
Good point. Done.